sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala

```diff
@@ -584,7 +584,6 @@ private[hive] class HiveClientImpl(
   }
 
   override def createTable(table: CatalogTable, ignoreIfExists: Boolean): Unit = withHiveState {
-    verifyColumnDataType(table.dataSchema)
```

Comment:
Any side-effect from this removal?

Reply:
Good point. @dongjoon-hyun 
Compared with the 3.5's, it's different as now we allow some special characters in nested struct field names that HMS recognizes to be stored.

Compared with the master, there is no difference as these special characters are regex-replaced to pass this check

Reply:
Got it.